src/overlays/actors/ovl_En_Demo_heishi/z_en_demo_heishi.c

```diff
@@ -31,37 +35,150 @@ const ActorInit En_Demo_heishi_InitVars = {
+
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 25.0f);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &gDemoheishiSkeleton, &gDemoheishiWave, this->jointTable,
+                       this->morphTable, 17);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.targetMode = 6;
+    this->actor.gravity = -3.0f;
+    Collider_InitAndSetCylinder(globalCtx, &this->colliderCylinder, &this->actor, &sCylinderInit);
+    EnDemoheishi_SetupIdle(this);
+}
+
+void EnDemoheishi_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnDemoheishi* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->colliderCylinder);
+}
+
+void EnDemoheishi_ChangeAnimation(EnDemoheishi* this, s32 animIndex) {
+    static AnimationHeader* sAnimations[] = {
+        &gDemoheishiStandHandOnHip, &gDemoheishiCheerWithSpear, &gDemoheishiWave,
+        &gDemoheishiSitAndReach,    &gDemoheishiStandUp,
+    };
+    static u8 sAnimModes[] = { 0, 0, 0, 0, 2, 0, 0, 0, 0, 0, 0, 0 };
+
+    this->animIndex = animIndex;
+    this->frameCount = Animation_GetLastFrame(sAnimations[animIndex]);
+    Animation_Change(&this->skelAnime, sAnimations[this->animIndex], 1.0f, 0.0f, this->frameCount,
+                     (u8)(s32)sAnimModes[this->animIndex], -10.0f);
+}
+
+void EnDemoheishi_SetHeadRotation(EnDemoheishi* this) {
+    s16 yawTemp = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+    s32 yaw = ABS_ALT(yawTemp);
+
+    this->headRotXTarget = 0;
+    if ((this->actor.xzDistToPlayer < 200.0f) && (yaw < 20000)) {
+        this->headRotXTarget = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+        if (this->headRotXTarget > 10000) {
+            this->headRotXTarget = 10000;
+        } else if (this->headRotXTarget < -10000) {
+            this->headRotXTarget = -10000;
+        }
+    }
+}
+
+void EnDemoheishi_SetupIdle(EnDemoheishi* this) {
+    EnDemoheishi_ChangeAnimation(this, 0);
+    this->actor.textId = sTextIds[this->unk272 = 0];
+    this->unk270 = 0;
+    this->actionFunc = EnDemoheishi_Idle;
+}
+
+void EnDemoheishi_Idle(EnDemoheishi* this, GlobalContext* globalCtx) {
+    s32 absYawDiff;
+    s16 yawDiff;
+
+    this->actor.flags &= ~0x8000000;
+    yawDiff = this->actor.yawTowardsPlayer - this->actor.world.rot.y;
+    absYawDiff = ABS_ALT(yawDiff);
+
+    if (func_800B84D0(&this->actor, globalCtx)) {
+        EnDemoheishi_SetupTalk(this);
+    } else if (absYawDiff <= 0x4BB8) {
+        func_800B8614(&this->actor, globalCtx, 70.0f);
+    }
+}
+
+void EnDemoheishi_SetupTalk(EnDemoheishi* this) {
+    this->unk270 = 1;
+    this->actionFunc = EnDemoheishi_Talk;
+}
+
+void EnDemoheishi_Talk(EnDemoheishi* this, GlobalContext* globalCtx) {
+    if ((func_80152498(&globalCtx->msgCtx) == 5) && (func_80147624(globalCtx) != 0)) {
+        func_801477B4(globalCtx);
+        EnDemoheishi_SetupIdle(this);
+    }
+}
+
+void EnDemoheishi_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ColliderCylinder* collider;
```

Comment:
```suggestion
    s32 pad;
```
Does it still match if this is just a normal `s32 pad` and delete the `collider = &this->colliderCylinder;` down below?

Reply:
It does! Thanks, it's been updated.